src/IReflection.php

```diff
@@ -0,0 +1,15 @@
+<?php
+/**
+ * Parser Reflection API
+ *
+ * @copyright Copyright 2015, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+namespace Go\ParserReflection;
+
+interface IReflection
```

Comment:
The purpose of this interface is intended to be more than it currently appears to be. Because we don't have multiple inheritance the idea was that each Reflection class would have an `IReflection...` interface, and that each of these would extend the parent of the original Reflection class. The purpose being that you could easily type-hint any parser-reflection class, with all compatible subtypes by using, for instance `IReflectionFunctionAbstract`, which would be implemented by all subtypes that are parser-reflection classes. I thought a full "Interface" prefix for this purpose would become too cumbersome. Also, native Reflections already have a `Reflector` interface, so, if I didn't want to go in this direction, I probably would have named this `Reflector` instead.  I also didn't want the compatibility headaches of renaming the interface later.

Interested to hear your feedback.

Reply:
...or perhaps you want me to rename this `Reflector`,  and add an empty derived interface (`IReflection` or `ReflectionInterface` or `Interfaces\Reflection` or whatever you think is best) when I add all its brethren? (Kicking the van down the road so to speak.)

Reply:
My POV is to name it as `ParserReflection\ReflectionInterface` and leave all stuff in this interface

Reply:
👍 

Still tempted to add `Go\ParserReflection\Reflector` in the future for consistency, but I see no need for that in this PR.

Reply:
@lisachenko, I'm finally digging into making these changes (as time is available), and a concern came to mind: It looks like some of the `ReflectionType` values in PHP 7.0 are being broken out into subclasses in PHP 7.2. If this trend continues, we may see `ReflectionTrait` and `ReflectionInterface` as derived classes of `ReflectionClass` in the not too distant future. If we wish to mirror this class heirarchy, this would cause a name collision for us. Perhaps we should change `IReflection` to `Reflector` which is already in PHP as an interface?

(Just a thought. In the meantime, I've already made the change you suggested. [Not pushed yet.] )